grafast/grafserv/src/servers/fastify/v4/index.ts

```diff
@@ -191,13 +187,49 @@ export class FastifyGrafserv extends GrafservBase {
         const result = await convertHandlerResultToResult(handlerResult);
         return this.send(request, reply, result);
       },
-      ...(this.resolvedPreset.grafserv?.websockets
-        ? {
-            wsHandler: makeHandler(makeGraphQLWSConfig(this)),
-          }
-        : null),
     });
 
+    const graphqlOverGet =
+      (dynamicOptions.graphqlOverGET || dynamicOptions.graphiqlOnGraphQLGET) ??
+      false;
+
+    const websockets = this.resolvedPreset.grafserv?.websockets ?? false;
+
+    if (graphqlOverGet || websockets) {
+      app.route({
+        method: "GET",
+        url: this.dynamicOptions.graphqlPath,
+        exposeHeadRoute: true,
+        ...(graphqlOverGet
+          ? {
+              handler: async (request, reply) => {
+                const digest = getDigest(request, reply);
+                const handlerResult = await this.graphqlHandler(
+                  normalizeRequest(digest),
+                  this.graphiqlHandler,
+                );
+                const result =
+                  await convertHandlerResultToResult(handlerResult);
+                return this.send(request, reply, result);
+              },
+            }
+          : {
+              handler: async (request, reply) => {
+                return reply.status(405).send({
+                  error: "Method not allowed.",
+                  detail:
+                    "Can be enabled with { grafserv: { graphqlOverGET: true } }",
+                });
+              },
```

Comment:
Looking at this, I realise that we already have this handling built in (search for `statusCode: 405`). So I propose an alternative solution:

```ts
const url = this.dynamicOptions.graphqlPath;
const bodyLimit = this.dynamicOptions.maxRequestLength;
const exposeHeadRoute = true;
const handler = async (request, reply) => {
  const digest = getDigest(request, reply);
  const handlerResult = await this.graphqlHandler(
    normalizeRequest(digest),
    this.graphiqlHandler,
  );
  const result = await convertHandlerResultToResult(handlerResult);
  return this.send(request, reply, result);
};
const wsHandler = websockets
  ? makeHandler(makeGraphQLWSConfig(this))
  : undefined;
app.route({ url, method: "POST", handler, bodyLimit });
if (websockets || graphqlOverGet) {
  app.route({ url, method: "GET", handler, exposeHeadRoute, wsHandler });
}
```